core/dbt/include/global_project/macros/adapters/columns.sql

```diff
@@ -75,11 +75,11 @@
      alter {{ relation.type }} {{ relation }}
 
             {% for column in add_columns %}
-               add column {{ column.name }} {{ column.data_type }}{{ ',' if not loop.last }}
+               add column "{{ column.name }}" {{ column.data_type }}{{ ',' if not loop.last }}
```

Comment:
See the commentary in https://github.com/dbt-labs/dbt-core/issues/4423#issuecomment-1116077864. We would want to use `adapter.quote` instead of just `"`, since the quote character can vary on different data platforms.

We'd also want a test for this change!